### ISSUE TITLE
A lot of "Updated Block: core/spacer" messages in the JS console in WP 5.9.1-RC1 w/ TT2

### ISSUE THREAD

**Issue by pbiron** [WordPress/gutenberg#38896]
### Description

During the 5.9.1-RC1 release party, I discovered that simply opening the site editor (with TT2) generates A LOT of messages in the JS console with `Updated Block: core/spacer` (see screenshot below).  By A LOT, I'm talking ~50 or so :-)

@audrasjb mentioned that this might have to do with https://github.com/WordPress/gutenberg/pull/38362

### Step-by-step reproduction instructions

1. Update to WP 5.9.1-RC1 (w/o GB active)
2. Activate the TT2 theme
3. Go to `Appearance > Editor`
4. Notice all the console messages

### Screenshots, screen recording, code snippet

![5 9 1-RC1-site-editor-console](https://user-images.githubusercontent.com/3824560/154577819-a7003bb3-1833-4e9b-9264-beec03f8cc6a.png)

### Environment info

WP 5.9.1-RC1 (w/o GB), TT2
Windows 10
Chrome 98.0.4758.102

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by pbiron** [comment]
I should also mention, they those messages do **not** appear when running WP 5.9.

**Comment by Mamaduka** [comment]
I think this is because of changing the `height` attribute from number to string (with units). The Deprecations code is a little more aggressive than we wanted.

@youknowriad, you resolved a similar issue with Cover block deprecation recently. Do you think we can apply the same method here as well?

**Comment by youknowriad** [comment]
> The Deprecations code is a little more aggressive than we wanted.

How so?

**Comment by Mamaduka** [comment]
Technically values returned from the save function haven't changed—only the attribute, from unitless number to string with the unit.

Deprecation is valid; I thought maybe we could ignore this one. Which I now realized isn't a good idea 😅 

P.S. This is the PR that introduced deprecation - https://github.com/WordPress/gutenberg/pull/36186.

**Comment by youknowriad** [comment]
The deprecation looks good but it seems it should only run if there's actually a width or height defined (not the default value). So I'm wondering from the screenshot above why the block is considered "invalid" (why the deprecation runs) since for the default values of height/width, the generated markup is still the same since it will use the new default value (with the units).

**Comment by t-hamano** [comment]
Perhaps it has something to do with the fact that the height of the spacers in the block pattern (theme / core / remote directory) is stored as a `number`, not a `string` ?

I have confirmed that stopping all block patterns as follows will stop all deprecated logs regarding core/spacer.

```php
// add_action( 'init', 'twentytwentytwo_register_block_patterns', 9 );

remove_theme_support( 'core-block-patterns' );
add_filter( 'should_load_remote_block_patterns', '__return_false' );
```

**Comment by t-hamano** [comment]
Related WordPress Ticket: 
[Twenty Twenty-Two: use custom units spacer blocks](https://core.trac.wordpress.org/ticket/55169)

**Comment by iandunn** [comment]
If these can't be removed, could they use `console.debug()` or `.info()` instead of `.log()`? That'd make it easy to hide them. Or maybe have some kind of flag that folks can set in `wp-config` to hide them?

Right now they're pretty distracting when I'm trying to check the console for relevant information. It gets drown out by all these messages.

**Comment by Mamaduka** [comment]
@youknowriad suggested that we disable info logging when pattern blocks are parsed - https://wordpress.slack.com/archives/C02QB2JS7/p1646643142026719

**Comment by fabiankaegy** [comment]
This was fixed in #39404
